org.eclipse.xtext/src/org/eclipse/xtext/linking/lazy/LazyLinker.java

```diff
@@ -277,7 +277,7 @@ protected EClass findSubTypeInEPackage(EPackage ePackage, EClass superType) {
 	}
 
 	private boolean isInstantiatableSubType(EClass c, EClass superType) {
-		return !c.isAbstract() && !c.isInterface() && EcoreUtil2.isAssignableFrom(superType, c);
+		return !c.isAbstract() && !c.isInterface() && (c == superType || EcoreUtil2.isAssignableFrom(superType, c));
```

Comment:
why here and not at org.eclipse.xtext.EcoreUtil2.isAssignableFrom(EClass, EClass)

Reply:
right, would be the better place
I wonder if we avoid the call from findInstantiableCompatible at all, it's just unneccessary